delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -0,0 +1,465 @@
+                        'type': port_type,
+                        'connection_status': conn_status,
+                        'health_status': status,
+                        'location': location,
+                        'storage_id': storage_id,
+                        'speed': max_speed,
+                        'max_speed': max_speed,
+                        'wwn': '',
+                        'mac_address': data.get('mac-address', ''),
+                        'ipv4': data.get('ip-address', '')
+                    }
+                    list_ports.append(dataMap)
+            return list_ports
+        except Exception as e:
+            err_msg = "Failed to get storage ports: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def list_storage_controller(self, storage_id):
+        try:
+            controller_info = self.ssh_pool\
+                .do_exec('show controllers')
+            controller_detail = self.handle_detail(controller_info,
+                                                   'controllers')
+            controller_arr = []
+            if controller_detail:
+                for data in controller_detail:
+                    health = data.get('health', '')
+                    status = constants.StoragePoolStatus.OFFLINE
+                    if health == 'OK':
+                        status = constants.StoragePoolStatus.NORMAL
+                    cpu_info = str(data.get('sc-cpu-type', ''))
+                    memory_size = data.get('system-memory-size') + "MB"
+                    system_memory_size = self.parse_string(
+                        memory_size)
+                    data_map = {
+                        'native_controller_id': data.get('controller-id', ''),
+                        'name': data.get('durable-id', ''),
+                        'storage_id': storage_id,
+                        'status': status,
+                        'location': data.get('position', ''),
+                        'soft_version': data.get('sc-fw', ''),
+                        'cpu_info': cpu_info,
+                        'memory_size': int(system_memory_size)
+                    }
+                    controller_arr.append(data_map)
+            return controller_arr
+        except Exception as e:
+            err_msg = "Failed to get storage controllers: %s"\
+                      % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def list_storage_volume(self, storage_id):
+        try:
+            volume_infos = self.ssh_pool.do_exec('show volumes')
+            volume_detail = self.handle_detail(volume_infos, 'volumes')
+            pools_info = self.ssh_pool.do_exec('show pools')
+            pool_detail = self.handle_detail(pools_info, 'pools')
+            list_volumes = []
+            if volume_detail:
+                for data in volume_detail:
+                    health = data.get('health', '')
```

Comment:
Default values of below  6 `ge` function is unnecessary.

Reply:
The default value is cancelled